var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -505,7 +505,8 @@ def cmake_args(self):
             cmake_args.append("-DLINK_POLLY_INTO_TOOLS:Bool=ON")
 
         if "+shared_libs" in spec:
-            cmake_args.append("-DBUILD_SHARED_LIBS:Bool=ON")
```

Comment:
Do you think this should be a conflict? Building `llvm_dylib` is really not the same, and unless it causes a build failure (and should be a conflict) I'm not sure about making it have no effect.

Reply:
Per the Document, yes; to quote it:

```
LLVM_BUILD_LLVM_DYLIB:BOOL
If enabled, the target for building the libLLVM shared library is added. This library contains all of LLVM’s components in a single shared library. Defaults to OFF. This cannot be used in conjunction with BUILD_SHARED_LIBS. Tools will only be linked to the libLLVM shared library if LLVM_LINK_LLVM_DYLIB is also ON. The components in the library can be customised by setting LLVM_DYLIB_COMPONENTS to a list of the desired components. This option is not available on Windows.
LLVM_LINK_LLVM_DYLIB:BOOL
If enabled, tools will be linked with the libLLVM shared library. Defaults to OFF. Setting LLVM_LINK_LLVM_DYLIB to ON also sets LLVM_BUILD_LLVM_DYLIB to ON. This option is not available on Windows.
BUILD_SHARED_LIBS:BOOL
Flag indicating if each LLVM component (e.g. Support) is built as a shared library (ON) or as a static library (OFF). Its default value is OFF. On Windows, shared libraries may be used when building with MinGW, including mingw-w64, but not when building with the Microsoft toolchain.
```

Is the`shared_libs` variant provided there just to set `BUILD_SHARED_LIBS` to ON? Is that it?

Reply:
Tom I was actually confused about those two variants.

Reply:
Yup, `shared_libs` builds with `BUILD_SHARED_LIBS` to cause all llvm binaries to link to shared versions of the libraries rather than static ones.  It reduces size and compile time substantially, which makes it useful for debugging or development work, but can cause subtle breakage since it isn't well tested in that configuration.  

The LLVM_DYLIB settings are whether the whole darned thing should be built as a single massive shared library, and if so whether the tools should use it.  They result in rather different final binaries.  Building the big llvm shared library can be required to build some other things.  The option to build the tools against it though are good for reducing binary size on disk IIRC, but bad for performance and optimization of the compiler itself, so I tend not to use it.

Reply:
Thanks for your comments. I'll withdraw this PR.